src/app/app.module.ts

```diff
@@ -91,68 +94,93 @@ export function appInitializer(
 
           // Set this back so that second flow through we do the proper code flow to get a refresh token.
           sessionStorage.setItem('syndesis-first-idp', 'false');
-          oauthService.hybrid = originalHybrid;
 
-          let autoLinkGithHub = configService.getSettings('oauth')[
-            'auto-link-github'
-          ];
-          if (autoLinkGithHub === undefined) {
-            autoLinkGithHub = true;
-          }
+          /**
+           * Redirects user to setup page if GitHub setup is still pending
+           */
+          const apiEndpoint = configService.getSettings().apiEndpoint;
+          const accessToken = oauthService.getAccessToken();
+          setupService.isSetupPending(apiEndpoint, accessToken)
```

Comment:
Wonder, could you not do `return setupService.isSetupPending(apiEndpoint, accessToken);` here and then add another promise handler?

That being said, maybe it's best to stick with nesting this call for starters, we can always come back and rework all this initialization code, I feel like as a whole it's gotten a bit unwieldy as we've needed to add functionality.